### PR TITLE
Enable icount for qemu_cortex_a53

### PR DESCRIPTION
**Pull request by andyross**
Seems to fix the timer glitchery on this platform.  Note twister patch that works around what seems to be an upstream qemu bug (and presumably the reason this wasn't already enabled).

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]

File: boards/arm64/qemu_cortex_a53/qemu_cortex_a53_defconfig [hunk 1]
```diff
@@ -1,7 +1,9 @@
 CONFIG_SOC_QEMU_CORTEX_A53=y
 CONFIG_BOARD_QEMU_CORTEX_A53=y
 CONFIG_ARM_ARCH_TIMER=y
-CONFIG_QEMU_ICOUNT=n
+CONFIG_QEMU_ICOUNT=y
+CONFIG_QEMU_ICOUNT_SHIFT=4
+CONFIG_QEMU_ICOUNT_SLEEP=y
```
Comment (stephanosio) [inline boards/arm64/qemu_cortex_a53/qemu_cortex_a53_defconfig:6]: It looks like icount was deliberately disabled for this platform:
https://github.com/zephyrproject-rtos/zephyr/commit/1be06335fa3bc0ad7655aff508950d62254b9f71

@carlocaione Is the problem mentioned in the above commit still relevant?

Also should https://github.com/zephyrproject-rtos/zephyr/pull/32049/commits/eadc73bfbe60284fcbe34432f472869869628528 be brought in again?

**Comment by povergoing** [comment]
1) I saw CI still fails with `qemu_cortex_a53_smp `board. Does it mean ICOUNT cannot fix it like X86_64?
2) With ICOUNT, I am not sure if it will cause testcase timeout since it would be slow, especially for those `syscall_torture` tests who badly break the simulation quantum ( That's why the FVP board added the `timeout_multiplier` )https://github.com/zephyrproject-rtos/zephyr/blob/b076aa373a63002617b81ac5813bf0668f0b6deb/boards/arm64/fvp_baser_aemv8r/fvp_baser_aemv8r_smp.yaml#L13-L14

File: boards/arm64/qemu_cortex_a53/qemu_cortex_a53_defconfig (hunk 1 above)
Comment (carlocaione) [reply boards/arm64/qemu_cortex_a53/qemu_cortex_a53_defconfig:6]: It was removed for performance reasons but I guess fixing the CI issues is more important.

File: boards/arm64/qemu_cortex_a53/qemu_cortex_a53_smp_defconfig [hunk 2]
```diff
@@ -2,6 +2,8 @@ CONFIG_SOC_QEMU_CORTEX_A53=y
 CONFIG_BOARD_QEMU_CORTEX_A53=y
 CONFIG_ARM_ARCH_TIMER=y
 CONFIG_QEMU_ICOUNT=n
```
Comment (carlocaione) [inline boards/arm64/qemu_cortex_a53/qemu_cortex_a53_smp_defconfig:4]: Do we want `CONFIG_QEMU_ICOUNT=y` here as well?

Comment (carlocaione) [reply boards/arm64/qemu_cortex_a53/qemu_cortex_a53_smp_defconfig]: Oh, this was just fixed.

**Review by carlocaione — DISMISSED** [review]

File: boards/arm64/qemu_cortex_a53/qemu_cortex_a53_smp_defconfig (hunk 2 above)
Comment (andyross) [reply boards/arm64/qemu_cortex_a53/qemu_cortex_a53_smp_defconfig]: Sorry, I forgot a note.  Yes, the smp platform was still showing failures so I removed the icount setting there; it's just voodoo anyway, SMP in qemu is always implemented with host threads, so it's always subject to host timer skew regardless of how the emulator works internally.

**Review by stephanosio — COMMENTED** [review]
Based on https://github.com/zephyrproject-rtos/zephyr/pull/48891/files#r943589948, re-enabling icount for the `qemu_cortex_a53` looks reasonable to me.

As for the twister change, based on the commit history, it does not seem like there was a specific reason for sticking with the `SIGTERM`. @nashif any thoughts?

By the way, any reason for changing the `handlers.py` permissions from 644 to 755?

**Comment by carlocaione** [comment]
I'm ok with merging this.

**Comment by fabiobaltieri** [comment]
This should solve the problem for the specific a53 NON-smp platform right? If it works I say it's a step forward and we should give it a go.

That said we still have the smp platforms issue. Trying to summarize the "known issue": we have a few timing dependent tests that are known to fail on qemu-smp platforms when qemu runs under heavy load on the host. In the current setup they reliably fail on the first run (because the machine/container is heavily loaded) but they reliably pass on the second run, so we have a hard dependency on the retry logic in the CI.

Does that sound right? We should probably document this somewhere, I'd say in `doc/develop/test/twister.rst` since this is got exposed on both kernel and networking tests. What do you think?

**Review by povergoing — DISMISSED** [review]
I am OK with it. 
Maybe we need a mechanism to set ICOUNT only for those time-sensitive test cases in the future?

**Comment by mbolivar-nordic** [comment]
> This should solve the problem for the specific a53 NON-smp platform right? If it works I say it's a step forward and we should give it a go.

Ironically, we can't do that right now, because CI is failing in kernel tests on `qemu_cortex_a53_smp` right now :(.

**Comment by andyross** [comment]
> Ironically, we can't do that right now, because CI is failing in kernel tests on qemu_cortex_a53_smp right now :(.

Oh, duh.  Yeah, that was the stale failure from before the dependency fix.  Just kicked it again.

**Review by fabiobaltieri — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by povergoing — APPROVED** [review]

**Pull request merged**